lib/recurly/BaseClient.js

```diff
@@ -106,6 +101,47 @@ class BaseClient {
 
     return options
   }
+
+  /**
+   * @param {Response} resp
+   * @return {(ApiError|null)}
+   */
+  _errorFromResponse (resp) {
+    if (resp.status < 200 || resp.status > 299) {
+      const errBody = resp.body && resp.body.error
+      // If we have a body, we determine the error based on
+      // the contents of the body
+      if (errBody) {
+        let className = utils.classify(errBody.type)
+        if (!className.endsWith('Error')) className += 'Error'
+        const ErrClass = apiErrors[className] || ApiError
+        const err = new ErrClass(errBody.message, errBody.type, errBody.params)
+        err._setResponse(resp)
+        return err
+      // if we don't have a body, we determine the error
+      // based on the http status code
+      } else {
+        let err = new ApiError('Unknown Error', 'unknown')
+
+        if (resp.status === 400) {
+          err = new apiErrors.BadRequestError('Bad Request', 'bad_request')
+        } else if (resp.status >= 401 && resp.status <= 403) {
+          err = new apiErrors.UnauthorizedError('Request not authorized. Perhaps your API key or site-id is incorrect?', 'unauthorized')
+        } else if (resp.status === 400) {
+          err = new apiErrors.NotFoundError('Resource Not Found', 'not_found')
+        } else if (resp.status === 422) {
+          err = new apiErrors.ValidationError('Request not valid', 'validation')
+        } else if (resp.status === 500) {
+          err = new apiErrors.InternalServerError('Internal Server Error', 'internal_server_error')
```

Comment:
Do we want to phrase this in a way that makes it clearer the error is on our side? Recurly Server Error?